src/main/java/apoc/sequence/Sequence.java

```diff
@@ -0,0 +1,151 @@
+        public LinkedList<Long> next(Long current, long length) {
+            return LongStream.range(current + 1, current + length + 1).boxed()
+                    .collect(Collectors.toCollection(LinkedList::new));
+        }
+
+        @Override
+        public String getName() {
+            return "LongCounter";
+        }
+
+    };
+
+    // Map <providerName, sequenceProvider> holding instances of provider for fast access.
+
+    private static final Map<String, SequenceProvider> SEQUENCE_PROVIDER_MAP = new ConcurrentHashMap<>();
+
+    static {
+        SEQUENCE_PROVIDER_MAP.put(LONG_SP.getName(), LONG_SP);
+    }
+
+    /**
+     * Registers a new provider. It will be available for sequence creation.
+     *
+     * @param provider
+     * @return True if the provider wasn't already registered, false otherwise.
+     */
+    public static boolean registerProvider(SequenceProvider provider) {
+        return SEQUENCE_PROVIDER_MAP.put(provider.getName(), provider) == null;
+    }
+
+    @Context
+    public GraphDatabaseService db;
+
+    @UserFunction(value = "apoc.sequence.get")
+    @Description("apoc.sequence.get('sequenceName') - returns the node representing that sequence, null if not existing.")
+    public Node get(@Name("sequenceName") String sequenceName) {
+        return db.findNode(SEQUENCE_LABEL, NAME_PROPERTY, sequenceName);
+    }
+
+
+    // TODO: 29/06/17 Can't be completely thread safe. Can't acquire a lock on a node if the node doesn't exist.
+    @Procedure(mode = Mode.WRITE)
+    @Description("apoc.sequence.create('sequenceName', 'providerName') - create sequence with given provider")
+    public Stream<NodeResult> create(@Name("sequenceName") String sequenceName,
+                                     @Name(value = "providerName", defaultValue = "LongCounter") String providerName) {
+        // MultipleFoundException if more than one.
+        Node sequenceNode = db.findNode(SEQUENCE_LABEL, NAME_PROPERTY, sequenceName);
+        if (sequenceNode != null)
+            throw new RuntimeException(String.format("There's already a sequence with name <%s> !", sequenceName));
+
+        SequenceProvider provider = SEQUENCE_PROVIDER_MAP.get(providerName);
+        if (provider == null)
+            throw new RuntimeException("No provider registered with name: " + providerName);
+
+        Node node = db.createNode(SEQUENCE_LABEL);
+        node.setProperty(NAME_PROPERTY, sequenceName);
+        node.setProperty(PROVIDER_PROPERTY, providerName);
+        node.setProperty(provider.propertyName(), provider.start());
+
+        return Stream.of(new NodeResult(node));
+    }
+
+    @Procedure(mode = Mode.WRITE)
+    @Description("apoc.sequence.next('sequenceName') - get&increment sequence value using related provider and length as increments number")
+    @SuppressWarnings("unchecked")
+    public Stream<ObjectResult> next(@Name("sequenceName") String sequenceName, @Name(value = "length", defaultValue = "1") long length) {
```

Comment:
I would probably limit the id-generation to `long` and not have object-results with boxing etc. What other sequence generators did you think about? The only others that make sense are UUIDs and MongoDB like object-ids which both need no centralized id-generation.